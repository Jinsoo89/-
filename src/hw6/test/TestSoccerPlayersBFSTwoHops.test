# Simple test of breadth-first search algorithm.
# Verifies that it can find a path with two "hops" (edges) when there are no
# other edges in the graph.
# Note that in this graph, there is an edge from C.Ronaldo to D.Maria but not
# vice versa. In your full parser-generated players graph, relationships are
# bidirectional, meaning an edge from A to B will always be accompanied by an
# edge from B to A.

CreateGraph players
AddNode players C.Ronaldo
AddNode players D.Maria
AddNode players L.Messi
AddEdge players C.Ronaldo D.Maria M.United
AddEdge players D.Maria L.Messi FC.Barcelona

FindPath players C.Ronaldo L.Messi

FindPath players D.Maria C.Ronaldo