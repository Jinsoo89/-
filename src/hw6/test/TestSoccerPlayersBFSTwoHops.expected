# Simple test of breadth-first search algorithm.
# Verifies that it can find a path with two "hops" (edges) when there are no
# other edges in the graph.
# Note that in this graph, there is an edge from C.Ronaldo to D.Maria but not
# vice versa. In your full parser-generated players graph, relationships are
# bidirectional, meaning an edge from A to B will always be accompanied by an
# edge from B to A.

created graph players
added node C.Ronaldo to players
added node D.Maria to players
added node L.Messi to players
added edge M.United from C.Ronaldo to D.Maria in players
added edge FC.Barcelona from D.Maria to L.Messi in players

path from C.Ronaldo to L.Messi:
C.Ronaldo to D.Maria via M.United
D.Maria to L.Messi via FC.Barcelona

path from D.Maria to C.Ronaldo:
no path found